client/blocks/credit-card/use-payment-processing.js

```diff
@@ -133,6 +133,11 @@ export const usePaymentProcessing = (
 					meta: {
 						paymentMethodData: {
 							stripe_source: response.source.id,
+							// The billing information here is relevant to properly create the
+							// Stripe Customer object.
+							billing_email: ownerInfo.email,
```

Comment:
If a logged in user with an email `foo@mail.dev` enters a different email in the checkout form like `foo2@mail.dev` then that email it's not considered and it's used the email the user used to register.

![image](https://user-images.githubusercontent.com/1025173/118720866-c380d880-b7ef-11eb-8b07-44d7db331c5f.png)

Is this intended behavior?

Reply:
I just want to mention the email used on the checkout form is indeed saved on Stripe. Since you posted screenshots

Reply:
I just want to mention the email used on the checkout form is indeed saved on Stripe and it's visible on the payment page.  Since you posted screenshots of the dashboard page on Stripe I thought the email used should be displayed there.

Reply:
I think this is a byproduct of how we fetch the Stripe Customer object; if someone is logged into an account on the site we use their account to retrieve a pre-existing Stripe Customer to associate with the payment instead of creating a new one just because the e-mail is different. This assumes they've checked out on that same account before.

I don't think it makes sense to change that behavior 🤔 
The order information sent to the customer uses the WC order information; the order information visible in wp-admin shows whatever the customer entered during Checkout, the description in the Stripe Dashboard is only different because there was already an existing Stripe Customer object associated with the customer's account.

Reply:
@reykjalin Thanks for clarifying. Are you going to fix the issue I mentioned on this PR? I think this is ready to go.

Reply:
I've fixed the issue you discovered in #1567, so I'm going to revert 1d3fe8e and instead rely on the fix in #1567 for this 🙂 

Nice catch!